lib/template/processors/svelte.js

```diff
@@ -0,0 +1,130 @@
+      if (hasProgessiveFlag) {
+        for (var i = 0; i < serverResult.vars.length; i++) {
+          var v = serverResult.vars[i];
+
+          // Exposing "public" would mean serializing the entire object to the client.
+          if (['partial', 'public'].includes(v.export_name)) {
+            throw new Error(`Cannot use "${v.export_name}" in progressive components`);
+          }
+        }
+      }
+
+      return function render(args) {
+        var rendered = Component.render(Object.assign({}, args, { enhanced: false }));
+
+        var clientScript = '';
+
+        // Compile the client script if needed.
+        if (hasProgessiveFlag) {
+          // E.g. `current`, but only if it is actually used.
+          var extraProps = {};
+
+          // For every exported component prop that is also a local, add it to the props.
+          // In other words: if a component uses a local such as `current` it will be serialized to the client.
+          serverResult.vars.forEach(function (v) {
+            // Exported component variable.
+            if (!v.module && v.export_name) {
+              if (
+                // Only expose these two for now, since I see value in both and they're trivial.
+                ['current', 'environment'].includes(v.export_name) &&
+                Object.prototype.hasOwnProperty.call(args, v.export_name)
+              ) {
+                extraProps[v.export_name] = args[v.export_name];
+              }
+            }
+          });
+
+          // TODO: In case we want to remove inline scripts and use external scripts,
+          // we can just generate a nanoid() and add that to the div
+          // instead of relying on the position (previousElementSibling) here.
+          var clientWrapper = `
+            import Component from ${devalue(options.filename)}
+
+            new Component({
+              // The <div> wrapper right before this script.
+              target: document.currentScript.previousElementSibling,
+              hydrate: true,
+              props: {
+                enhanced: true,
+                ...(${devalue(extraProps)})
+              }
+            });
+          `;
+
+          clientScript = esbuildClient({
+            stdin: {
+              contents: clientWrapper,
+              resolveDir: path.dirname(options.filename),
+              sourcefile: 'harp-svelte-inline-wrapper.js',
+            },
+            bundle: true,
+            minify: true,
+            write: false,
+          }).outputFiles[0].text;
+        }
+
+        var styleBlock = '';
+
+        if (serverResult.css.code) {
+          styleBlock = `<style>${rendered.css.code}</style>`;
+        }
+
+        var scriptBlock = '';
+
+        if (clientScript) {
+          scriptBlock = `<script>${clientScript}</script>`;
+        }
+
+        return `<div style="display: contents;">${rendered.html}</div>${scriptBlock}${styleBlock}`;
```

Comment:
FWIW this is probably still open to XSS because the `devalue` above in the inline script will run through esbuild making it kind of pointless. Depending on what esbuild does this might re-enable `</script>` inside any string. So either we need to replace that here or if we go 100% external scripts then no worries